src/utils/aepp-wallet-communication/rpc/rpc-client.ts

```diff
@@ -0,0 +1,308 @@
+/**
+ * RpcClient module
+ *
+ * @module @aeternity/aepp-sdk/es/utils/aepp-wallet-communication/rpc/rpc-client
+ * @export { RpcClient, RpcClients }
+ * @example
+ * import RpcClient from '@aeternity/aepp-sdk/es/utils/aepp-wallet-communication/rpc/rpc-client'
+ */
+
+import { METHODS, RPC_STATUS, SUBSCRIPTION_TYPES } from '../schema'
+// @ts-expect-error TODO remove
+import { sendMessage, message, isValidAccounts } from '../helpers'
+import {
+  InvalidRpcMessageError,
+  TypeError,
+  DuplicateCallbackError,
+  MissingCallbackError
+} from '../../errors'
+
+export interface Connection {
+  isConnected: () => boolean
+  disconnect: (forceConnectionClose?: boolean) => void
+  connect: (
+    handleMessage: (msg: any, origin: string) => void,
+    disconnect: (connection: Connection) => void
+  ) => void
+  connectionInfo: {id: string}
+  id: string
+  origin?: string
+  debug: boolean
+  forceOrigin: boolean
+  sendDirection?: string
+  receiveDirection: string
+  subscribeFn: Function
+  unsubscribeFn: Function
+  postFn: Function
+  listener?: Function
+}
+
+export interface RpcClientInfo {
+  name?: string
+  icons?: [{ src: string, sizes?: string, type?: string, purpose?: string }]
+  status?: string
+  connectNode?: boolean
+  version?: number
+  origin?: string
+}
+
+export interface Accounts {
+  connected?: { [pub: string]: {} }
+  current?: { [pub: string]: {} }
+}
+
+export interface Message {
+  jsonrpc: string
+  id: number
+  method: string
+  version: number
+  params?: any
+  result?: any
+  error?: {
+    code: number
+    data?: any
+    message: string
+  }
+}
+
+/**
+ * Contain functionality for using RPC conection
+ * @alias module:@aeternity/aepp-sdk/es/utils/aepp-wallet-communication/rpc/rpc-client
+ * @function
+ * @rtype Stamp
+ * @param {Object} param Init params object
+ * @param {String} param.name Client name
+ * @param {Object} param.connection Connection object
+ * @param {Function[]} param.handlers Array with two function for message handling
+ * @param {Function} param.handlers[0] Message handler
+ * @param {Function} param.handlers[1] Disconnect callback
+ * @return {Object}
+ */
+
+export default class RpcClient {
+  id: string
+  connection: Connection
+  info: RpcClientInfo
+  callbacks: Map<string | number, {resolve: Function, reject: Function}>
+  addressSubscription: string[]
+  accounts: Accounts
+  sendMessage: Function
+  addresses: string[]
+  currentAccount: string
```

Comment:
```suggestion
  readonly accounts: Accounts
  sendMessage: Function
  readonly addresses: string[]
  readonly currentAccount: string
```

as these fields are defined with `configurable: false`, they could also be defined as `readonly`